tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -692,608 +505,989 @@ void fillShaderBufferFromLongVectorData(std::vector<BYTE> &ShaderBuffer,
+//
+// AsTypeOp
+//
 
-  if (BasicOpType != BasicOpType_Unary && BasicOpType != BasicOpType_Binary)
-    LOG_ERROR_FMT_THROW(L"Programmer Error: computeExpectedValue called with "
-                        L"unexpected BasicOpType: %d",
-                        static_cast<int>(BasicOpType));
+void dispatchAsUintSplitDoubleTest(const TestConfig &Config,
+                                   size_t VectorSize) {
 
-  if (ExpectedValueComputer)
-    ExpectedVector = ExpectedValueComputer->computeExpectedValues(Inputs);
-  else
-    // Only SplitDouble has special handling. All other ops will have an
-    // ExpectedValueComputer set.
-    computeExpectedValues_SplitDouble(Inputs.InputVector1);
-}
+  InputSets<double, 1> Inputs = buildTestInputs<double, 1>(Config, VectorSize);
 
-template <typename T>
-void AsTypeOpTestConfig<T>::computeExpectedValues_SplitDouble(
-    const std::vector<T> &InputVector) {
-
-  DXASSERT_NOMSG(OpType == AsTypeOpType_AsUint_SplitDouble);
-
-  // SplitDouble is a special case. We fill the first half of the expected
-  // vector with the expected low bits of each input double and the second
-  // half with the high bits of each input double. Doing things this way
-  // helps keep the rest of the generic logic in the LongVector test code
-  // simple.
-  std::vector<uint32_t> Values;
-  Values.resize(InputVector.size() * 2);
-
-  uint32_t LowBits, HighBits;
-  const size_t InputSize = InputVector.size();
-
-  for (size_t Index = 0; Index < InputSize; ++Index) {
-    splitDouble(InputVector[Index], LowBits, HighBits);
-    Values[Index] = LowBits;
-    Values[Index + InputSize] = HighBits;
+  std::vector<uint32_t> Expected;
+  Expected.resize(Inputs.size() * 2);
+
+  for (size_t I = 0; I < Inputs.size(); ++I) {
+    uint32_t Low, High;
+    splitDouble(Expected[I], Low, High);
+    Expected[I] = Low;
+    Expected[I + Inputs.size()] = High;
   }
 
-  ExpectedVector = std::move(Values);
+  ValidationConfig ValidationConfig{};
+  runAndVerify(Config, AsTypeOpType_AsUint_SplitDouble, Inputs, Expected,
+               " -DFUNC_ASUINT_SPLITDOUBLE=1", ValidationConfig);
 }
 
-template <typename T>
-TrigonometricOpTestConfig<T>::TrigonometricOpTestConfig(
-    const OpTypeMetaData<TrigonometricOpType> &OpTypeMd)
-    : TestConfig<T>(OpTypeMd), OpType(OpTypeMd.OpType) {
+void dispatchTest(const TestConfig &Config, AsTypeOpType OpType,
+                  size_t VectorSize) {
 
-  static_assert(
-      isFloatingPointType<T>(),
-      "Trigonometric ops are only supported for floating point types.");
+  // Different AsType* operations are supported for different data types, so we
+  // dispatch on operation first.
 
-  BasicOpType = BasicOpType_Unary;
+#define DISPATCH(TYPE, FN)                                                     \
+  if (Config.DataType == DataTypeName<TYPE>())                                 \
+  return dispatchUnaryTest<TYPE>(Config, ValidationConfig{}, OpType,           \
```

Comment:
this needs indented another level right? Not sure if the diff is just being weird

Reply:
No, the diff is correct.  However, clang-format gets final say here and this is what it wants.